zeebe/gateway-rest/src/test/java/io/camunda/zeebe/gateway/rest/controller/usermanagement/GroupControllerTest.java

```diff
@@ -463,6 +466,140 @@ void shouldReturnErrorForAddingUserToMissingGroup() {
     verify(groupServices, times(1)).assignMember(groupId, username, EntityType.USER);
   }
 
+  @Test
+  void shouldAssignMappingToGroupAndReturnAccepted() {
+    // given
+    final var groupId = Strings.newRandomValidIdentityId();
+    final var mappingId = Strings.newRandomValidIdentityId();
+    when(groupServices.assignMember(groupId, mappingId, EntityType.MAPPING))
+        .thenReturn(CompletableFuture.completedFuture(null));
+
+    // when
+    webClient
+        .put()
+        .uri("%s/%s/mapping-rules/%s".formatted(GROUP_BASE_URL, groupId, mappingId))
+        .accept(MediaType.APPLICATION_JSON)
+        .exchange()
+        .expectStatus()
+        .isAccepted();
+
+    // then
+    verify(groupServices, times(1)).assignMember(anyString(), anyString(), any());
+  }
+
+  @Test
+  void shouldReturnErrorForAddingMissingMappingToGroup() {
+    // given
+    final var groupId = Strings.newRandomValidIdentityId();
+    final var mappingId = Strings.newRandomValidIdentityId();
+    when(groupServices.assignMember(groupId, mappingId, EntityType.MAPPING))
+        .thenReturn(
+            CompletableFuture.failedFuture(
+                new CamundaBrokerException(
+                    new BrokerRejection(
+                        RoleIntent.ENTITY_ADDED,
+                        1L,
+                        RejectionType.NOT_FOUND,
+                        "Mapping rule not found"))));
+
+    // when
+    webClient
+        .put()
+        .uri("%s/%s/mapping-rules/%s".formatted(GROUP_BASE_URL, groupId, mappingId))
+        .accept(MediaType.APPLICATION_PROBLEM_JSON)
+        .exchange()
+        .expectStatus()
+        .isNotFound();
+
+    // then
+    verify(groupServices, times(1)).assignMember(anyString(), anyString(), any());
+  }
+
+  @Test
+  void shouldReturnErrorForAddingMappingToMissingGroup() {
+    // given
+    final var groupId = Strings.newRandomValidIdentityId();
+    final var mappingId = Strings.newRandomValidIdentityId();
+    when(groupServices.assignMember(groupId, mappingId, EntityType.MAPPING))
+        .thenReturn(
+            CompletableFuture.failedFuture(
+                new CamundaBrokerException(
+                    new BrokerRejection(
+                        RoleIntent.ENTITY_ADDED, 1L, RejectionType.NOT_FOUND, "Role not found"))));
```

Comment:
🔧 Copy-paste error. The rejection refers to roles instead of groups which could cause confusion in the future.